### ISSUE TITLE
what is 'gate' used for ?

### ISSUE THREAD

**Issue by kangkang59812** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#68]
In models.py:
`gate = self.sigmoid(self.f_beta(h[:batch_size_t])) `
`attention_weighted_encoding = gate * attention_weighted_encoding`

The 'attention_weighted_encoding' has been weighted by alpha,right? So, what is 'gate' used for ?

**Comment by kangkang59812** [comment]
By the way, where can I find the processing code for `Andrej Karpathy's training, validation, and test splits`. I want to create the file for COCO2017

**Comment by TranTony** [comment]
@kangkang59812 Due to the input of the gate is the previous hidden_state, I think it should be used to 'emphasize' into the hidden_state.

**Comment by kmario23** [comment]
> In models.py:
> `gate = self.sigmoid(self.f_beta(h[:batch_size_t])) `
> `attention_weighted_encoding = gate * attention_weighted_encoding`
> 
> The 'attention_weighted_encoding' has been weighted by alpha,right? So, what is 'gate' used for ?

This is called as `doubly stochastic regularization` in the section 4.2.1 of the [SAT paper](http://proceedings.mlr.press/v37/xuc15.pdf). Quoting directly from the paper:

> This gating variable lets the decoder decide whether to put more emphasis on language modeling or on the context at each time step

**Comment by kangkang59812** [comment]
@TranTony @kmario23 thanks a lot!!!
